internal-dns/src/config.rs

```diff
@@ -465,13 +456,18 @@ impl DnsConfigBuilder {
             .chain(srv_records_zones)
             .collect();
 
+        DnsConfigZone { zone_name: DNS_ZONE.to_owned(), records: all_records }
+    }
+
+    /// Construct a complete [`DnsConfigParams`] (suitable for propagating to
+    /// our DNS servers) for the control plane DNS zone described up to this
+    /// point
+    pub fn build(self) -> DnsConfigParams {
+        let zone = self.build_zone();
         DnsConfigParams {
-            generation: u64::from(self.generation),
+            generation: u64::from(Generation::new()),
```

Comment:
Yeah, fair question.  The history as I recall it is:

- This struct was originally only used to build the _initial_ DNS configuration so the generation was always 1.  I believe it's used in this way by both RSS and the automated test runner setup stuff.
- A few weeks ago I wanted to use this struct in nexus-reconfigurator-execution to build a new DNS configuration, too.  There, I needed to produce a different generation, so I added a function to specify the generation and it would use that.
- Now for the reasons mentioned in the description I'm changing the caller in nexus-reconfigurator-execution so that it only gets the _zone_ out of this object.  So I'm removing the (now unused) method to change the generation and hardcoding it back to 1, since the only callers that get a `DnsConfigParams` here want generation 1.  We could always add the other thing back but at this point it does not appear needed.

I think of it as: this thing used to build a `DnsConfigParams`, but now really generates a `DnsConfigZone`, with a convenience method for assembling that into a `DnsConfigParams` with generation 1 for the callers that want that.  I thought about renaming it to `build_full_config_for_initial_generation()` or something to better distinguish it from `build_zone()` but I didn't bother -- wasn't sure the extra mouthful was really clearer.  But since you had this question, I went ahead and did this.  (I don't think it makes sense to accept the generation here given that all existing callers provide generation 1.  It's not at all hard to build your own `DnsConfigParams` if you want a different one and the whole point of this convenience function is to commonize code among the 5 or so callers so if they all have to decide which generation to put here it defeats part of the point.)

Reply:
Perfect, thanks. I do think the longer name is worth the clarity.